lib/internal/modules/package_json_reader.js

```diff
@@ -154,10 +175,50 @@ function getPackageType(url) {
   return getPackageScopeConfig(url).type;
 }
 
+const pjsonImportAttributes = { __proto__: null, type: 'json' };
+let cascadedLoader;
+/**
+ * @param {URL['href'] | URL['pathname']} specifier The location for which to get the "root" package.json
+ * @param {URL['href'] | URL['pathname']} parentLocation The location of the current module (ex file://tmp/foo.js).
+ */
+function findPackageJSON(specifier, parentLocation) {
+  validateString(specifier, 'specifier');
```

Comment:
We should probably coerce the `specifier` to a string, like `import()` does, so it's possible to pass e.g. a `URL` instance.
```suggestion
  specifier = `${specifier}`;
```

Reply:
I think it might be better UX to check if it is a URL and if not, validate that it's a string (better error messaging, and fails faster/earlier).